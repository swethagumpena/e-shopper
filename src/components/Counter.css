.counter{
    display: flex
}

button{
    margin: 2em;
}