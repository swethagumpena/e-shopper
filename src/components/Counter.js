import React from 'react';
import PropTypes from 'prop-types';
import styles from './Counter.module.css';

function Counter(props) {
  const { value, onIncrement, onDecrement } = props;
  return (
    <div className={styles.counter}>
      <button onClick={onDecrement} type="button">-</button>
      <p className={`${styles.container} ${value === 0 ? styles.containerQuantity0 : ''}`}>
        {value}
        {' '}
        in Basket
      </p>
      <button onClick={onIncrement} type="button">+</button>
    </div>
  );
}

Counter.propTypes = {
  value: PropTypes.number.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Counter;
