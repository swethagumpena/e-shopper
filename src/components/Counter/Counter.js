import React from 'react';
import PropTypes from 'prop-types';
import styles from './Counter.module.css';

const Counter = ({ value, onIncrement, onDecrement }) => (
  <div className={styles.counter}>
    <button onClick={onDecrement} type="button">-</button>
    {/* <p className={`${styles.container} ${value === 0 ? styles.containerQuantity0 : ''}`}> */}
    <p className={(value === 0 ? styles.containerQuantity0 : styles.container)}>
      {value}
      {' '}
      in Basket
    </p>
    <button onClick={onIncrement} type="button">+</button>
  </div>
);

Counter.propTypes = {
  value: PropTypes.number.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Counter;
