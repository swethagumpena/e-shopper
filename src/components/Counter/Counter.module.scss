@import '../../styles/index.scss';

.counter{
    display: flex;
    align-items: center;
    justify-content: center;
    p {
        padding: 0;
    }
}

// .counter p {
//     padding: 0;
// }

.container {
    font-weight: normal;
    background-color:khaki;
}

.containerQuantity0{
    background-color: #D0D0D0;
}