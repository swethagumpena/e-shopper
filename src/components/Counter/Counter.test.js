/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import Counter from './Counter';

describe('example', () => {
  test('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
});

describe(Counter.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const mockProps = {
    value: 2,
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };

  test('should match snapshot', () => {
    const { container } = render(<Counter {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  test('should display the current count', () => {
    render(<Counter {...mockProps} />);
    expect(screen.getByText('2 in basket'));
  });

  // render(<Counter
  //   value={2}
  //   onIncrement={() => {}}
  //   onDecrement={() => {}}
  // />)

  test('should display an increment button', () => {
    render(<Counter {...mockProps} />);
    const incrementElement = screen.getByText('+');
    expect(incrementElement.tagName).toBe('BUTTON');
  });

  test('should display a decrement button', () => {
    render(<Counter {...mockProps} />);
    const incrementElement = screen.getByText('-');
    expect(incrementElement.tagName).toBe('BUTTON');
  });

  test('should increment product count', () => {
    render(<Counter {...mockProps} />);
    const incrementElement = screen.getByText('+');
    fireEvent.click(incrementElement);
    expect(mockProps.onIncrement).toHaveBeenCalled();
  });

  test('should decrement product count', () => {
    render(<Counter {...mockProps} />);
    const decrementElement = screen.getByText('-');
    fireEvent.click(decrementElement);
    expect(mockProps.onDecrement).toHaveBeenCalled();
  });
});
