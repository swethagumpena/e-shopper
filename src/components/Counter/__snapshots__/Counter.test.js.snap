// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div>
  <div
    class="counter"
  >
    <button
      type="button"
    >
      -
    </button>
    <p
      class="container"
    >
      2 in basket
    </p>
    <button
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
